src/frontends/tensorflow/src/op/slice.cpp

```diff
@@ -18,11 +18,22 @@ OutputVector translate_slice_op(const NodeContext& node) {
     auto start = node.get_input(1);
     auto size = node.get_input(2);
 
-    auto stop = make_shared<Add>(start, size);
+    // compute stop values in case non-negative sizes
+    auto stop_pos = make_shared<Add>(start, size);
 
-    auto one = make_shared<Constant>(element::i64, Shape{1}, 1);
-    auto shape = make_shared<ShapeOf>(start);
-    auto step = make_shared<Broadcast>(one, shape);
+    // compute stop values in case positive sizes
+    auto input_shape = make_shared<ShapeOf>(input);
+    auto one = make_shared<Constant>(size.get_element_type(), Shape{}, 1);
+    auto stop_neg = make_shared<Add>(make_shared<Add>(make_shared<ConvertLike>(input_shape, size), size), one);
+
+    // select the correct stop value based on a sign of size value
+    auto zeros = make_shared<Constant>(size.get_element_type(), Shape{}, 0);
+    auto negative_sizes_mask = make_shared<Less>(size, zeros);
+    auto stop = make_shared<Select>(negative_sizes_mask, stop_neg, stop_pos);
+
+    // broadcast step value
+    auto start_shape = make_shared<ShapeOf>(start);
+    auto step = make_shared<Broadcast>(one, start_shape);
 
```

Comment:
How this would behave if the first dimension of the input is 0? For example: (0, X, Y).

Reply:
`start_shape` is 1D tensor.
In case input to be empty tensor, it will compute `stop` to be equal to `start` by this dimension and the output will be also empty tensor.